File: src/co/aospa/sense/camera/callables/StopPreviewCallable.java
package co.aospa.sense.camera.callables;

import android.hardware.Camera;

import co.aospa.sense.camera.listeners.CameraListener;

public class StopPreviewCallable extends CameraCallable {
    private static final String TAG = StopPreviewCallable.class.getSimpleName();

    public StopPreviewCallable(CameraListener cameraListener) {
        super(cameraListener);
    }

    @SuppressWarnings("deprecation")
    @Override
    public CallableReturn call() {
        Camera camera = getCameraData().mCamera;
        if (camera == null) {
            return new CallableReturn(new Exception("Camera isn't opened"));
        }
        camera.stopPreview();
        return new CallableReturn(null);
    }

    @Override
    public String getTag() {
        return TAG;
    }
}

File: src/co/aospa/sense/camera/callables/StopPreviewCallable.kt
package co.aospa.sense.camera.callables

import co.aospa.sense.camera.listeners.CameraListener
import java.lang.Exception

class StopPreviewCallable(listener: CameraListener?) : CameraCallable(listener) {

    override fun call(): CallableReturn {
        val camera = cameraData.mCamera
            ?: return CallableReturn(Exception("Camera isn't opened"))
        camera.stopPreview()
        return CallableReturn(null)
    }

    override val tag: String?
        get() = StopPreviewCallable::class.java.simpleName
}